tests/test_openapi.py

```diff
@@ -239,6 +240,32 @@ def test_field_with_allow_none(self, openapi):
         else:
             assert res['nullable'] is True
 
+    def test_field_with_str_regex(self, openapi):
+        regex_str = '^[a-zA-Z0-9]$'
+        field = fields.Str(validate=validate.Regexp(regex_str))
+        ret = openapi.field2property(field)
+        assert ret['pattern'] == regex_str
+
+    def test_field_with_pattern_obj_regex(self, openapi):
+        regex_str = '^[a-zA-Z0-9]$'
+        field = fields.Str(validate=validate.Regexp(re.compile(regex_str)))
+        ret = openapi.field2property(field)
+        assert ret['pattern'] == regex_str
+
+    def test_field_with_no_pattern(self, openapi):
+        field = fields.Str()
+        ret = openapi.field2property(field)
+        assert 'pattern' not in ret
+
+    def test_field_with_multiple_patterns(self, openapi):
+        regex_validators = [
+            validate.Regexp('loser'),
+            validate.Regexp('winner'),
```

Comment:
I'm not sure about this "last one wins" behavior, but I can't think of a better alternative at the moment.

Perhaps we should raise a warning if there are multiple `Regexp` validators on a field.

Reply:
Yeah, apispec doesn't really enforce correctness of the spec. There are other similar instances of undefined or invaild output (https://github.com/marshmallow-code/apispec/issues/198).

AFAIU, only one pattern can be specified (I could imagine validating with several patterns but I don't think OpenAPI allows that.). Let's assume the user knows what he is doing and he only specified one regex and we can write

```py
        validator = next(
            (v for v in field.validators if hasattr(v, 'regex') and isinstance(v.regex, RegexType)),
            None
        )
        if validator:
            attributes['pattern'] = v.regex.pattern
        return attributes
```

Or we go further and we add a warning if several regex are specified.

I think both are better than the "last one wins" method.

Reply:
Sure, code updated to build a generator expression. If can be iterated more than once a UserWarning gets raised.